qiskit/transpiler/passes/optimization/optimize_1q_decomposition.py

```diff
@@ -174,4 +174,27 @@ def run(self, dag):
                 for current_node in run[1:]:
                     dag.remove_op_node(current_node)
 
+        # Runs for parameters gate
+        runs = dag.collect_1q_runs_with_params()
+        for run in runs:
+            gate = run[0]
+            new_ops = []
+            for ngate in run[1:]:
+                if gate.op.name in {"rz", "ry", "rx"} and gate.op.name == ngate.op.name:
+                    # merge parameters
+                    gate.op.params[0] = gate.op.params[0] + ngate.op.params[0]
+                else:
+                    new_ops.append(gate.op)
+                    gate = ngate
+            new_ops.append(gate.op)
+
+            idx = 0
+            for op in new_ops:
+                dag.substitute_node(run[idx], op)
+                idx += 1
+
+            # Delete the other nodes in the run
+            for current_node in run[idx:]:
+                dag.remove_op_node(current_node)
+
```

Comment:
This is a nice idea. If I understand correctly, currently the code completely replaces chains of parameterized "rz", "ry", "rx" gates by "optimized" chains. Would it be possible to avoid this replacement if no (or only few) optimizations take place?

Reply:
This worked in that way previously. But in the code, the optimization is done by converting the gates to the matrix and multiplying the unitary of the gates. And at the end, the decomposition is done. So before my changes the parameters gates were not optimized.